src/bluecore_api/app/routes/works.py

```diff
@@ -69,8 +69,8 @@ async def update_work(
 
     # Update data if it is provided
     if work.data is not None:
```

Comment:
There's no need to parse the JSON into a graph and frame it because it happens automatically when .data is being assigned to.